docs/src/developer/codebase.md

```diff
@@ -0,0 +1,34 @@
+# Navigating the codebase
+
+The code is organised using a Rust workspace with seprate crates for the node and faucet binaries, a crate for each node
+component, a couple of gRPC-related codegen crates, and a catch-all utilities crate.
+
+The primary artifacts are the node and faucet binaries. The library crates are not intended for external usage, but
+instead simply serve to enforce code organised and decoupling. The node binary is used to launch the components i.e. we
+have a single binary with subcommands for each component (and running them all as a single process).
+
+```mermaid
+---
+title: Workspace dependency tree
+---
+graph TD;
+    
+  subgraph components
+      block-producer
+      store
+      rpc
+  end
+
+  faucet ---> proto & utils;
+  node --> components & utils
+
+  components --> proto & utils
+
+  proto; utils;
+```
```

Comment:
I would maybe just have a simple table with crate name and description - kind of similar to how we do it in [miden-vm](https://github.com/0xPolygonMiden/miden-vm?tab=readme-ov-file#project-structure) readme  and maybe complement it with an excalidraw diagram.